src/app/loadout/plug-drawer/SelectablePlug.tsx

```diff
@@ -21,47 +24,100 @@ export default function SelectablePlug({
   selected,
   selectable,
   selectionType,
+  stack,
   removable,
   onPlugSelected,
   onPlugRemoved,
 }: {
   plug: PluggableInventoryItemDefinition;
+  /** How many of this specific mod have been selected? (only for stackable mods) */
+  stack: number;
   classType: DestinyClass;
   selected: boolean;
   selectable: boolean;
-  selectionType: 'multi' | 'unique' | 'single';
+  selectionType: PlugSelectionType;
   removable: boolean;
   onPlugSelected: (plug: PluggableInventoryItemDefinition) => void;
   onPlugRemoved: (plug: PluggableInventoryItemDefinition) => void;
 }) {
-  const handleClick = useCallback(() => {
-    selectable && onPlugSelected(plug);
-  }, [onPlugSelected, plug, selectable]);
+  const canRemoveOnClick =
+    selectionType !== PlugSelectionType.Single && selected && removable && stack <= 1;
 
-  const onClose = useCallback(() => onPlugRemoved(plug), [onPlugRemoved, plug]);
+  const handleClick = useCallback(() => {
+    selectable ? onPlugSelected(plug) : canRemoveOnClick && onPlugRemoved(plug);
+  }, [selectable, onPlugSelected, plug, canRemoveOnClick, onPlugRemoved]);
 
   // Memoize the meat of the component as it doesn't need to re-render every time
   const plugDetails = useMemo(
     () => <SelectablePlugDetails plug={plug} classType={classType} />,
     [plug, classType],
   );
 
+  const stackable = stack > 1 || (stack === 1 && selectable);
+
+  const handleIncrease =
+    stackable &&
+    selectable &&
+    ((e: React.MouseEvent) => {
+      e.stopPropagation();
+      onPlugSelected(plug);
+    });
+
+  const handleDecrease =
+    stackable &&
+    removable &&
+    ((e: React.MouseEvent) => {
+      e.stopPropagation();
+      onPlugRemoved(plug);
+    });
+
   return (
-    <ClosableContainer onClose={selected && removable ? onClose : undefined}>
-      <TileGridTile
-        selected={selected}
-        disabled={selectionType !== 'single' && !selectable}
-        title={plug.displayProperties.name}
-        icon={
+    <TileGridTile
+      selected={selected}
+      disabled={
+        !(
+          selectable ||
+          canRemoveOnClick ||
+          (selectionType === PlugSelectionType.Single && selected)
+        )
+      }
+      title={plug.displayProperties.name}
+      icon={
+        <div className={styles.icon}>
           <div className="item" title={plug.displayProperties.name}>
             <DefItemIcon itemDef={plug} />
           </div>
-        }
-        onClick={handleClick}
-      >
-        {plugDetails}
-      </TileGridTile>
-    </ClosableContainer>
+          {(handleIncrease || handleDecrease) && (
+            <div className={styles.buttons}>
+              {handleIncrease && (
+                <button
+                  type="button"
+                  className="dim-button"
+                  onClick={handleIncrease}
+                  title={t('LB.AddStack')}
```

Comment:
This could just be the preview site maybe but I am seeing this

<img width="117" alt="Screenshot 2023-12-21 at 9 35 15 pm" src="https://github.com/DestinyItemManager/DIM/assets/7344652/a42acf64-aec9-4abb-a0d0-72e553b293cc">

The preview site is really cool btw